File: tests/UITests/PageObjectModels/TheInternetPages/CheckboxesPage.js
const { BasePage } = require('./BasePage');

class CheckboxesPage extends BasePage {
  constructor(page) {
    super(page);
    this.url = '/checkboxes';
    this.checkboxes = page.getByRole('checkbox');
  }

  async navigate() {
    await this.navigateTo(this.url);
  }

  async isCheckboxChecked(index) {
    return this.checkboxes.nth(index).isChecked();
  }

  async toggleCheckbox(index) {
    await this.checkboxes.nth(index).click();
  }
}

module.exports = { CheckboxesPage };

File: tests/UITests/PageObjectModels/TheInternetPages/CheckboxesPage.cs
using Microsoft.Playwright;

namespace Tests.UITests.PageObjectModels.TheInternetPages;

public class CheckboxesPage : BasePage
{
    private const string Url = "https://the-internet.herokuapp.com/checkboxes";

    private ILocator Checkboxes => Page.GetByRole(AriaRole.Checkbox);

    public CheckboxesPage(IPage page) : base(page) { }

    public async Task Navigate() => await NavigateTo(Url);

    public async Task<bool> IsCheckboxChecked(int index) => await Checkboxes.Nth(index).IsCheckedAsync();

    public async Task ToggleCheckbox(int index) => await Checkboxes.Nth(index).ClickAsync();
}
